content/tutorials/presence.textile

```diff
@@ -31,49 +34,118 @@ Presence enables clients to be aware of other clients that are currently “pres
+  Create a blank HTML page called @example.html@, to host your web app:
+
+  ```[html]
+      <!DOCTYPE html>
+      <html lang="en">
+        <head>
+          <meta charset="UTF-8" />
+          <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+          <title>Ably Presence</title>
+        </head>
+        <body>
+        </body>
+      </html>
+  ```
 
-  Include the code below just before your closing your HTML @</html>@ tag.
+blang[javascript].
+  To start using Ably in your web app, you first need to include the Ably client library. In this example, we recommend that you include the latest client library from our CDN using a simple @<script>@ tag. You should then instantiate the client library with the API key you copied in Step 1.
 
-  ```[javascript]
-    <!-- Include the latest Ably Library  -->
-    <script src="//cdn.ably.io/lib/ably.min-1.js"></script>
+  bq. **Note**: in production, Ably recommends that you always use the "token authentication scheme":/core-features/authentication#token-authentication for browser clients, however in this example you will use an API key for simplicity.
+
+blang[javascript].
+  Include the code below just before your closing @</html>@ tag. 
 
-    <!-- instantiate the Ably library  -->
-    <script type="text/javascript">
-      var realtime = new Ably.Realtime({key: apiKey, clientId: 'jamie'});
+  ```[javascript]
+    <script src="https://cdn.ably.com/lib/ably.min-1.js"></script>
+    <script>
+      const apiKey = "YOUR_ABLY_API_KEY"; // Replace with your API key
+      const randomId = Math.random().toString(36).slice(-10); 
+      const realtime = new Ably.Realtime.Promise({
+        key: apiKey,
+        clientId: randomId, // Your ID in the presence set
+      });
     </script>
   ```
 
-  "See this step in Github":https://github.com/ably/tutorials/commit/presence-javascript-step2
-
 blang[nodejs].
   To start using Ably you first need to install the NPM module. The NPM module can be installed as follows:
 
   ```[sh]
     npm install ably
   ```
 
-  The client library must be instantiated with the API key you copied in Step 1. API keys used with "basic authentication":/core-features/authentication#basic-authentication for your own servers is generally preferred, however clients running on insecure devices should always use the "token authentication scheme":/core-features/authentication#token-authentication instead. In this example, we use an API key for simplicity.
+  The client library must be instantiated with the API key you copied in Step 1. API keys used with "basic authentication":/core-features/authentication#basic-authentication for your own servers is generally preferred, however clients running on insecure devices should always use the "token authentication scheme":/core-features/authentication#token-authentication instead. In this tutorial, you will use an API key for simplicity.
```

Comment:
Same as before, adding just in case links for Step 1.

Reply:
As above